### DISCUSSION TITLE
Is this project dead?

### DISCUSSION THREAD

**Discussion by stevekatsaras** [fomantic/Fomantic-UI#2251] [General] [answered]
Hi all,
Is this project dead? I don't mean to be disrespectful to anyone, but I was an avid user of semantic-ui for many years till I stumbled upon a thread in GitHub that mentioned that semantic-ui was dead (as a project) and to migrate to fomantic-ui. But now I fear that this project has suffered the same fate. There are many issues being posted that contain little to no replies, discussion threads are old and unanswered. So this makes my curious and suspicious. Should we continue to use fomantic-ui? There are little to no releases of new versions!

Some guidance would be helpful.

**Comment by lubber-de** [comment, marked as answer]
fomantic UI is actively maintained. The active branch is develop instead of master. That's why official stable releases aren't that often.
However we provide a nightly beta on npm/cdn including the latest features of 2.9.0 .

This is a community fork, everybody is welcome to help out to speed up development. 😉

Most of the open issues which are tagged as "next-release" have already been fixed in the nightly beta. We are closing them once 2.9.0 gets stable. (There is still a lot to do before officially releasing it), so i suggest you use the beta version to help testing/finding (fixing😉) bugs.

The discussion feature is unfortunately not much used by the community, so you might join our [discord server](https://discord.gg/YChxjJ3) where I feel there is more action taking place.

**Reply by stevekatsaras** [reply]
Thanks @lubber-de .
I asked because I see very little progress in development and releases and very little chatter in the community to engage and help answer people's questions. It would be good to see small incremental releases more often, that would be great.

What kind of skills do you need for people to help out in the development space?
It would be great to see some kind of IDE, builder studio for Fomantic-UI, where you can drag'n'drop widgets on a canvas to build a page would be awesome.

**Reply by hugopeek** [reply]
> It would be great to see some kind of IDE, builder studio for Fomantic-UI, where you can drag'n'drop widgets on a canvas to build a page would be awesome.

@stevekatsaras Something like this? https://github.com/fomantic/Fomantic-UI/discussions/1917 :)

**Comment by lubber-de** [comment]
Thanks for the honest opinion, @stevekatsaras 

>  little progress

You may take a look at the [commit history](https://github.com/fomantic/Fomantic-UI/commits/develop) of the develop branch or the [list or merged PRs](https://github.com/fomantic/Fomantic-UI/pulls?q=is%3Apr+sort%3Aupdated-desc+is%3Amerged) or the [changelog of the past 3,5 years](https://github.com/fomantic/Fomantic-UI/blob/develop/CHANGELOG.md)
I would not call this "little progress" considering this is made from a few people doing everything unpaid in their free time.
Now compare that to the commit history of SUI... we are almost 3,5 years and 1500 commits ahead. 

>  very little chatter

I personally like, when people are focussing on a related topics, instead of argueing about (unrelated) stuff. 😆 
Again, our [discord](https://discord.gg/YChxjJ3) Server community most probably responds faster than on github discussions (if there is somebody available to answer your question)

> What kind of skills do you need for people to help out in the development space?

Let me quote myself from https://github.com/fomantic/Fomantic-UI/issues/319#issuecomment-786758096 :

> Well, it's a **community** fork: Please try to help and fix things yourself and provide a PR afterwards. If you are able to code JS to use fomantic, you most probably are also able to fix some things yourself and donate your work to the community. 😉 

The only skills people need to help improving FUI is.... a proper understanding of Javascript, CSS and HTML which basically everybody using FUI should already have 😉 
Some issues are tagged with "help-wanted" where people can look at (most of them are related to gulp/npm/yarn/non maintained 3rd party dependencies, etc.)
Whenever people are regularly providing a few good PRs we possibly ask them to become an official maintainer (where we would expect donating some of their free time for also reviewing PRs, answer questions )... that was happening to me in 2018 😄 

> It would be great to see some kind of IDE, builder studio

There are tools like https://www.sketch.com/ and some people once started to develop [FUI](https://github.com/obsidiansystems/fomantic-sketch-ui-kit) / [SUI](https://github.com/stoneddesigner/semantic-ui-sketch) sketch files for that. I never use those, so i don't follow their progress or the WYSIWYG market at all.

**Reply by stevekatsaras** [reply]
Thanks for the reply @lubber-de .
I completely understand the nature of open-source and community driven initiatives, I wasn't intending on putting it down. I guess all I was trying to say was it would be nice to see those smaller, frequent, incremental releases ... it shows activity.

I do know some JS, CSS, HTML. I have used Fomantic-UI in my project, but I am not quite sure how good I would be in fixing and contributing to the codebase. I am mainly a back-end Java/Spring developer, with some front-end experience.

With sketch.com, that looks interesting, but how does that integrate with Fomantic-UI ?

**Reply by lubber-de** [reply]
> I guess all I was trying to say was it would be nice to see those smaller, frequent, incremental releases ... it shows activity.

I basically agree. However, the next release [2.9.0](https://github.com/fomantic/Fomantic-UI/milestone/27?closed=1) contains slightly breaking changes. I believe, inside the 2.9.x version branch, releases will take place more often again 😉 

> With sketch.com, that looks interesting, but how does that integrate with Fomantic-UI 

It won't. It's just an independent tool to create HTML code out of a WYSIWYG drag/drop Editor, afaik.

**Reply by y0hami** [reply]
@stevekatsaras Just wanted to jump in here.

> but I am not quite sure how good I would be

Lots of people feel this way and its the main reason people don't help with open source. A good way to help would be just try and identify the issue and post it on the related issue and someone will take a look and if its correct will push the changes. This way you help us by doing the work but don't need to put in all the time of actually doing it and creating a PR and if it turns out its wrong no harm is done :)

**Reply by stevekatsaras** [reply]
thanks @hammy2899 . so, if i were interested in fixing bugs, how best would i go about setting everything up? Is there a process to follow? Is there a guide? I would not know which bugs to fix and/or what branch to cut a feature branch out of. Some instructions/guidance would be welcome.

Many thanks.

**Reply by y0hami** [reply]
Take a look at our [contributing guide](https://github.com/fomantic/Fomantic-UI/blob/develop/CONTRIBUTING.md) and once you're setup you can look for any issue or take a look at the once market with help-wanted or good-first-issue 

Take a look here for a list of good issues to start with https://github.com/fomantic/Fomantic-UI/issues?q=is%3Aopen+is%3Aissue+label%3Atag%2Fhelp-wanted

**Comment by hugopeek** [comment]
Hi @stevekatsaras,

Just wanted to chime in here too, as I've been using SUI / FUI for quite some time now. _SUI_ is definitely dead, so I switched to FUI a few years ago. Most of the pressing issues were being fixed very rapidly at the time, so there was a lot of energy and momentum behind this project. The discussions are still there about a possible version 3, but I don't know if that is being actively worked on anymore.

The way I see it today, as @lubber-de has pointed out already, is that maintaining a project this size requires a lot of effort. The reason why SUI development stagnated, is (in my view) because the maintainer never opened up the project to maintainers. So the issues and PRs kept piling up. Now, there are a couple of maintainers, and although the project could definitely use a couple more, things have been progressing steadily over the past years. The group has set up a proper workflow for triaging bugs and reviewing PRs, and they have always been very responsive to my queries and PRs. For that, they deserve a lot of credit. Which you can also provide through sponsorship ;)

They are also strict with adhering to semantic versioning (I mean, how can you not be if you forked a semantic UI ;)), which means a lot of corrections have been held back because they might break things (although, as pointed out, these are not earth shattering changes). I agree that a 2.8.9 release would have been good, while the work on 2.9 is still ongoing. Alternatively, you can also switch to the nightly releases, if you accept the obvious risks of unintentional side effects. I've been doing that for a while now in some projects, without much hassle.

Building stuff is easy / exciting / cool. Maintaining stuff is hard / boring / geeky. But I rather stick to what I know, because constantly switching frameworks and libraries also means constantly changing your workflow and replacing one set of problems for another. The Semantic UI philosophy still stands today, so I hope this project will keep that practically alive.
